workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-etc-clean-up/src/main/java/com/liferay/ticket/TicketCleanUpCommandLineRunner.java

```diff
@@ -0,0 +1,175 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ticket;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.OAuth2AuthorizeRequest;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * @author Gregory Amerson
+ */
+@Component
+public class TicketCleanUpCommandLineRunner implements CommandLineRunner {
+
+	@Override
+	public void run(String... args) throws Exception {
+		OAuth2AuthorizedClient oAuth2AuthorizedClient =
+			_authorizedClientServiceOAuth2AuthorizedClientManager.authorize(
+				OAuth2AuthorizeRequest.withClientRegistrationId(
+					"liferay-ticket-etc-clean-up-oauth-application-headless-server"
+				).principal(
+					"TicketCleanUpCommandLineRunner"
+				).build());
+
+		if (oAuth2AuthorizedClient == null) {
+			_log.error("Unable to get OAuth 2 authorized client");
+
+			return;
+		}
+
+		OAuth2AccessToken oAuth2AccessToken =
+			oAuth2AuthorizedClient.getAccessToken();
+
+		if (_log.isInfoEnabled()) {
+			_log.info("Issued: " + oAuth2AccessToken.getIssuedAt());
+			_log.info("Expires At: " + oAuth2AccessToken.getExpiresAt());
+			_log.info("Scopes: " + oAuth2AccessToken.getScopes());
+			_log.info("Token: " + oAuth2AccessToken.getTokenValue());
+		}
+
+		WebClient.Builder builder = WebClient.builder();
+
+		WebClient webClient = builder.baseUrl(
+			_lxcDXPServerProtocol + "://" + _lxcDXPMainDomain
+		).defaultHeader(
+			HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE
+		).defaultHeader(
+			HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE
+		).defaultHeader(
+			HttpHeaders.AUTHORIZATION,
+			"Bearer " + oAuth2AccessToken.getTokenValue()
+		).build();
+
+		TicketsResponse ticketsResponse = webClient.get(
+		).uri(
+			"/o/c/tickets/"
```

Comment:
Should we do the filtering of the tickets here with a filter query parameter? Instead of further down when we filter in the java stream?